arc_cache.go

```diff
@@ -63,32 +65,37 @@ func (b *arccache) hasCached(k *cid.Cid) (has bool, size int, ok bool) {
 	h, ok := b.arc.Get(k.KeyString())
 	if ok {
 		b.hits.Inc()
-		if h.(int) > -1 {
-			return true, h.(int), true
-		} else {
-			return false, h.(int), true
+		switch h := h.(type) {
+		case cacheHave:
+			return bool(h), -1, true
+		case cacheSize:
+			return true, int(h), true
 		}
 	}
 	return false, -1, false
 }
 
 func (b *arccache) Has(k *cid.Cid) (bool, error) {
```

Comment:
@taylormike 

I rushed this change through as I ended up leaving a bunch of packages semi-broken (due to https://github.com/ipfs/go-datastore/pull/92).

Before, `arccache.Has` would always call GetSize on the underlying datastore. Now, it calls `Has` unless someone as already called `GetSize` or `Get` (to avoid the overhead of actually getting the value). Will this change interfere with your bitswap changes?

Reply:
@Stebalien 
No, this will not interfere with my bitswap changes. I tested your recent changes and my bitswap changes together. They both look good. 
Thank you for your help and the quick turnaround.